### ISSUE TITLE
[9.0.0-rc.2] onRest and fn for useTransition states does not work 

### ISSUE THREAD

**Issue by zenios** [react-spring/react-spring#1000]
onRest in the following example is never called

```
const transition = useTransition(isOpen, {
    from: {
      opacity: 0,
    },
    enter: (active) => {
      return {
        opacity: 1,
        onRest: () => {
          console.log("Enter completed");
        },
      };
    },
    leave: {
      opacity: 0,
    },
  });
```

In these one it works
```
const transition = useTransition(isOpen, {
    from: {
      opacity: 0,
      onRest: () => {
        console.log("Hello 1");
      },
    },
    enter: {
      opacity: 1,
      onRest: () => {
        console.log("Enter completed")
      },
    },
    leave: {
      opacity: 0,
    },
  });
```

**Comment by aleclarson** [comment]
Bug confirmed: https://codesandbox.io/s/react-spring-1000-bq7wu

**Comment by aleclarson** [comment]
Fixed in `9.0.0-rc.3`

https://codesandbox.io/s/fixed-react-spring-1000-bzlng?file=/src/App.tsx
